### ISSUE TITLE
cockpit-client

### ISSUE THREAD

**Issue by jelly** [cockpit-project/cockpit#17014]
### Explain what happens

1. Try to login with a url or $domain:9090 of which the port is active
2. Cockpit-client hangs forever trying to connect?

### Version of Cockpit

262

### Where is the problem in Cockpit?

Unknown or not applicable

### Server operating system

Arch Linux

### What browsers are you using?

Cockpit Client / Cockpit Desktop

**Comment by garrett** [comment]
Part of the problem is not knowing that you are connecting via SSH, not HTTPS.

But still, hanging is an issue and not communicating what's expected and what's wrong are problems.

**Comment by garrett** [comment]
Messaging was included in previous mockups @ https://github.com/cockpit-project/cockpit/issues/16672#issuecomment-983782267

![image](https://user-images.githubusercontent.com/10246/154249840-54b30d46-d7c2-4756-8c5f-fe3438b233cb.png)

This wireframe mockup and alternates are available @:

[🡳 **cockpit-client-hosts-alt.excalidraw.zip**](https://github.com/cockpit-project/cockpit/files/7635418/cockpit-client-hosts-alt.excalidraw.zip)
https://excalidraw.com/#json=0w2nTfxh8Xtr4HCu7L6-A,u5YCUgRJRMeBNC_Yd24xgw

**Comment by garrett** [comment]
We should also probably use something similar to PF [Form](https://www.patternfly.org/v4/components/form) HelperText on the connect to entry or change the label to something like "Connect via ssh to:"

We could use a placeholder too, as long as it follows guidelines. 

Both are mentioned @ https://www.patternfly.org/v4/components/form/design-guidelines/#placeholder-text-0:

> ### Placeholder text
> 
> Use placeholder text to demonstrate correct inputs so that users are aware of formatting expectations and specifications needed to provide information without errors.
> 
> If your placeholder text demonstrates a correct input, introduce it as an example so that users can distinguish impermanent examples from field labels and their own inputted text.
> 
> Examples:
> 
> - *Example, John Smith*
> - *Example, (555) 555-5555*
> 
> Placeholder text that doesn’t give an example doesn’t need to be introduced as one. If your placeholder text simply communicates a field's required format, write the format without an introduction.
> 
> Examples:
> 
> - *MM-DD-YYYY*
> - *johnsmith@example.com*
> 
> **Note:** Once users start typing, placeholder text disappears. Consider communicating crucial formatting requirements or directions with helper text instead.
> 
> See our [text input guidelines](https://www.patternfly.org/v4/components/text-input/design-guidelines) for more specific guidance on writing placeholder text for text input fields.
> 
> ### Helper text
> 
> Use helper text to display critical secondary information that helps users input correct information on their first attempt. Describe any conditions for their inputted information to prevent unnecessary errors, like specifying that a name for a new asset must be a unique one.

**Comment by jelly** [comment]
I'm thinking we should have a cancel button when you are trying to login, and that reverts back to the "login screen".

**Comment by garrett** [comment]
> I'm thinking we should have a cancel button when you are trying to login, and that reverts back to the "login screen".

Yep, that would be great. The place for implementing that would be the login screen, I guess? (Some of Cockpit Client's functionality is on the login page, with some exclusives for Client that aren't exposed in normal web Cockpit.)
